timely/src/progress/timestamp.rs

```diff
@@ -9,10 +9,13 @@ use crate::communication::Data;
 use crate::order::PartialOrder;
 
 /// A composite trait for types that serve as timestamps in timely dataflow.
+///
+/// By implementing this trait, you promise that the type's [PartialOrder] implementation
+/// is compatible with [Ord], such that if `a.less_equal(b)` then `a <= b`.
 pub trait Timestamp: Clone+Eq+PartialOrder+Debug+Send+Any+Data+Hash+Ord {
     /// A type summarizing action on a timestamp along a dataflow path.
     type Summary : PathSummary<Self> + 'static;
-    /// A minimum value suitable as a default.
+    /// A unique minimum value in our partial order, suitable as a default.
```

Comment:
Reading this, I can't imagine what "suitable as a default" means. I know I wrote it, but .. reminder that we should probably rip it out.